src/edges_analysis/analysis/levels.py

```diff
@@ -468,7 +468,21 @@ def read_step(
     if fname.suffix == ".acq":
         return io.FieldSpectrum(fname).data
     else:
-        return _ReductionStep._get_object_class(fname)(fname)
+        return get_step_type(fname)(fname)
+
+
+def get_step_type(
+    fname: tp.PathLike | _ReductionStep | io.HDF5RawSpectrum,
+) -> _ReductionStep | io.HDF5RawSpectrum:
```

Comment:
Ah, there should be a `Type[]` around the output types (i.e. `-> Type[_ReductionStep] | Type[io.HDF5RawSpectrum]`)

Reply:
fixed